nengo/simulator.py

```diff
@@ -135,18 +131,15 @@ class Simulator:
 
     def __init__(
             self, network,
-            dt=0.001, seed=None, model=None, progress_bar=True, optimize=True,
-            dtype=global_dtype):
+            dt=0.001, seed=None, model=None, progress_bar=True, optimize=True):
```

Comment:
The thing I don't like about this is that it makes the dtype harder to change within a file, for example if I want to make one simulator with float32 and one with float64 (for example, to look at how that effects performance). This is still possible by setting it directly on the `Model`, though, so I'm fine to have that as a backdoor way to do this, since it'll mostly be power users or developers wanting this feature.

The other reason I added the Simulator `dtype` originally is that I really don't like scripts that are not self-contained (i.e. rely on having certain RC file settings, etc.). I think it's much better to put such parameters in the script itself. I know it's possible to configure RC settings from within a script (see `test_dtype` here), but we don't actually have an example in [the docs](https://www.nengo.ai/nengo/nengorc.html). We should add an example or two and promote that as a way to use the system. Would there be any potential problems with changing the `dtype` right after Nengo import with this commit? It actually looks like it's pretty safe. This wasn't something I was thinking about when I first did this, and obviously the reason you switched to `rc.dtype`. We just have to make sure we don't use `rc.dtype` as a function argument default anywhere, I think. It might actually get tricky when we rebase this onto master, though, since there the `NdarrayParam` has a `dtype`.

Reply:
As I mention in the commit message, I don't like having two different but interacting ways to specify a thing. I think in general we are wary of the RC system so we half use it, but we should just use it when we use it.

I can add some examples of changing RC settings in the docs, if that's important to you.

This is already rebased onto master.

Reply:
Those examples don't have to happen now, just wanted to point that out.

I don't think that `NdarrayParam.dtype` will change in the case where someone changes the `rc.dtype` after importing nengo, since params get created on objects at import. The two solutions are make `NdarrayParam.dtype` a property and have it get `rc.dtype` if it's underlying value is `None`, or just not care. Thoughts?

Reply:
I did the property thing.